server/src/main/java/org/opensearch/action/admin/indices/segment_replication/TransportSegmentReplicationAction.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.indices.segment_replication;
+
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.DefaultShardOperationFailedException;
+import org.opensearch.action.support.broadcast.node.TransportBroadcastByNodeAction;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.block.ClusterBlockException;
+import org.opensearch.cluster.block.ClusterBlockLevel;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.ShardsIterator;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.index.IndexService;
+import org.opensearch.index.shard.IndexShard;
+import org.opensearch.indices.IndicesService;
+import org.opensearch.indices.replication.SegmentReplicationState;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Transport action for shard segment replication operation. This transport action does not actually
+ * perform segment replication, it only reports on metrics of segment replication event (both active and complete).
```

Comment:
This looks to be following the pattern from the recovery API example, but maybe you could head off the obvious confusion this comment attempts to clear up by choosing a different name for all these classes? Like "TransportSegmentReplicationStatsAction" or something? Whatever is chosen would apply to all the related classes here.

Reply:
Yes makes sense, it would be clear if all these classes have "SegmentReplicationStats" instead of just "SegmentReplication". I will update this in next commits